WordPress/gutenberg
# Divider: Complete TypeScript migration of component

## What?
Completes the migration of `Divider` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactored test to use TypeScript
- Updated docs
- Sorted props in readme and types alphabetically 

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Divider` continues to function as expected
3. Test the various Dividers within the block/site editors in Gutenberg still work
